### PR TITLE
[model] feat: add ltx2-3 model

### PR DESCRIPTION
**Pull request by JimmyMrr**
### Checklist Before Starting

- Search for relative PRs/issues and link here: ...
- PR title follows `[{modules}] {type}: {description}` format (enforced by [check_pr_title.yml](.github/workflows/check_pr_title.yml))
  - **Allowed modules:** `agent`, `ci`, `ckpt`, `config`, `data`, `dist`, `docker`, `docs`, `logging`, `lora`, `misc`, `model`, `omni`, `optim`, `ops`, `parallel`, `perf`, `release`, `task`, `trainer`
  - **Allowed types:** `feat`, `fix`, `refactor`, `chore`, `test`
  - Breaking changes: prepend `[BREAKING]` — e.g. `[BREAKING][parallel, model] feat: dynamic batching`

step | LTX-2 src code | VeOmni | Comparison absolute
-- | -- | -- | --
1/100 | 0.4767 | 0.4779 | 0.0012
2/100 | 0.4759 | 0.4773 | 0.0014
3/100 | 0.4738 | 0.4749 | 0.0011
4/100 | 0.4684 | 0.4695 | 0.0011
5/100 | 0.4595 | 0.4603 | 0.0008
... | ...  |  ... | ...
96/100 | 0.0131 | 0.0131 | 0.0000
97/100 | 0.0130 | 0.0131 | 0.0001
98/100 | 0.0130 | 0.0131| 0.0001
99/100 | 0.0130 | 0.0131| 0.0001
100/100 | 0.0129 | 0.0130 | 0.0001

```
veomni/
├── models/diffusers/ltx2_3/   # LTX-2.3 模型实现
│   ├── ltx_core/              # 核心组件 (从 ltx-core 迁移)
│   │   ├── model/             # Transformer, VAE
│   │   ├── text_encoders/     # Gemma 文本编码器
│   │   └── components/        # Patchifier 等
│   ├── ltx_condition/         # 条件模型
│   ├── ltx_transformer/       # Transformer 包装器
├── trainer/dit_trainer.py     # DiT 训练器 (扩展)
├── data/dataset.py            # 数据集 (扩展)
└── configs/dit/ltx2_*.yaml    # 训练配置
```

### Checklist Before Submitting

- Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md)
- Applied pre-commit checks
- Added/updated documentation
- If `tasks/` training scripts were moved or renamed: updated `docs/` examples and verified `python3 scripts/ci/check_doc_task_paths.py` passes (also enforced by the **Check doc task paths** CI workflow)
- Added tests to CI workflow (or explained why not feasible)

### CODE REVIEW

File: veomni/trainer/dit_trainer.py [hunk 1]
```diff
@@ -433,6 +488,50 @@ def postforward(
         loss = torch.stack(list(loss_dict.values())).sum()
         return loss, loss_dict
 
+    def _compute_dit_loss(
```
Comment (FoolPlayer) [inline veomni/trainer/dit_trainer.py]: It's better not add ltx logic in general Dit-trainer, can you try to add these ltx logic in ltx condition/models? And for preprocess ltx data, we can add a ltx special data transform / collator, bu not adding in dit-trainer directly.

File: veomni/models/diffusers/ltx2_3/ltx_condition/modeling_ltx2_3_condition.py [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+        return normalized_means
+
+    @torch.no_grad()
+    def get_condition(self, inputs, videos, **kwargs) -> dict[str, Any]:
+        """Online encoding: Gemma + feature extractor for text, VAE for video.
+
+        Note: For offline precompute, use the precompute_gemma_features.py script instead.
+        This method requires Gemma to be loaded, which is expensive.
+        """
+        raise NotImplementedError(
```
Comment (FoolPlayer) [inline veomni/models/diffusers/ltx2_3/ltx_condition/modeling_ltx2_3_condition.py:265]: It looks it will get error if `online training`, but `ltx2_av_lora_low_vram.yaml` and `ltx2_v2v_ic_lora.yaml` use online training, it will get error.

File: configs/dit/ltx2_v2v_ic_lora.yaml [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    # These are the conditioning inputs that guide the transformation
+    reference_latents_dir: "reference_latents"
+    # --- Flow matching ---
+    timestep_sampling_mode: "shifted_logit_normal"
+
+# -----------------------------------------------------------------------------
+# Training Configuration
+# -----------------------------------------------------------------------------
+train:
+  training_task: online_training
```
Comment (FoolPlayer) [inline configs/dit/ltx2_v2v_ic_lora.yaml]: Will raise NotImplementedError due to LTXVideoConditionModel.get_condition() is not supported.

File: configs/dit/ltx2_av_lora_low_vram.yaml [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    first_frame_conditioning_p: 0.5
+    with_audio: true
+    audio_latents_dir: "audio_latents"
+    timestep_sampling_mode: "shifted_logit_normal"
+
+# -----------------------------------------------------------------------------
+# Training Configuration
+# -----------------------------------------------------------------------------
+train:
+  training_task: online_training
```
Comment (FoolPlayer) [inline configs/dit/ltx2_av_lora_low_vram.yaml]: Same as ltx2_v2v_ic_lora.yam

File: configs/dit/ltx2_av_lora_low_vram.yaml [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+# -----------------------------------------------------------------------------
+# Model Configuration
+# -----------------------------------------------------------------------------
+model:
+  # Path to the LTX-2.3 transformer weights
+  model_path: "path/to/ltx-2.3-model.safetensors"
+
+  # Path to the condition model (text encoder + scheduler + VAE)
+  condition_model_path: "path/to/gemma-text-encoder"
```
Comment (FoolPlayer) [inline configs/dit/ltx2_av_lora_low_vram.yaml]: It's conflict with ltx2_av_lora.yaml and L33 comment, maybe it should be ltx ckpt path?

File: configs/dit/ltx2_v2v_ic_lora.yaml [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+# -----------------------------------------------------------------------------
+# Model Configuration
+# -----------------------------------------------------------------------------
+model:
+  # Path to the LTX-2.3 transformer weights
+  model_path: "path/to/ltx-2.3-model.safetensors"
+
+  # Path to the condition model (text encoder + scheduler + VAE)
+  condition_model_path: "path/to/gemma-text-encoder"
```
Comment (FoolPlayer) [inline configs/dit/ltx2_v2v_ic_lora.yaml]: Same as `ltx2_av_lora_low_vram.yaml` setting.

File: veomni/models/diffusers/ltx2_3/ltx_transformer/modeling_ltx2_3_transformer.py [hunk 7]
```diff
@@ -0,0 +1,458 @@
...
+    audio_args = self.audio_args_preprocessor.prepare(audio, video) if audio is not None else None
+
+    if get_parallel_state().sp_enabled and video_args is not None:
+        video_args_x = slice_input_tensor(video_args.x, dim=1, group=get_parallel_state().sp_group)
+
+        cos_freq, sin_freq = video_args.positional_embeddings
+        ulysses_size = get_parallel_state().ulysses_size
+        ulysses_rank = get_parallel_state().ulysses_rank
+        seq_len = cos_freq.shape[2]
+        chunk = seq_len // ulysses_size
```
Comment (FoolPlayer) [inline veomni/models/diffusers/ltx2_3/ltx_transformer/modeling_ltx2_3_transformer.py]: Use genernal sp api to slice. like `slice_input_tensor`.

**Comment by FoolPlayer** [comment]
And it seems there is no experience tests for low-vram/v2v task, I think we can split this PR to some small PRs, in this one, just make sure the basic training is OK.

File: veomni/models/diffusers/ltx2_3/ltx_condition/preprocess_dataset.py [hunk 8]
```diff
@@ -0,0 +1,1868 @@
...
+
+    # Preprocess with reference videos for IC-LoRA
+    python preprocess_dataset.py preprocess \\
+        --dataset_file /path/to/dataset.json \\
+        --gemma_model_path /path/to/gemma3 \\
+        --checkpoint_path /path/to/ltx2.safetensors \\
+        --resolution_buckets 768x768x49 \\
+        --reference_column reference_path
+
+Output directory structure (after full pipeline):
```
Comment (Coach257) [inline veomni/models/diffusers/ltx2_3/ltx_condition/preprocess_dataset.py:68]: Update these docs. It seams the saved dataset is data_dir/ xxx.parquet

**Comment by Coach257** [comment]
Maybe add doc docs/examples

File: veomni/models/diffusers/ltx2_3/ltx_transformer/modeling_ltx2_3_transformer.py [hunk 9]
```diff
@@ -0,0 +1,510 @@
...
+
+        self.config = hf_config
+
+    @classmethod
+    def from_pretrained(cls, path, **kwargs):
+        from ....loader import get_model_config
+
+        kwargs.pop("trust_remote_code", None)
+        config = get_model_config(path, **kwargs)
+        model = cls._from_config(config)
```
Comment (FoolPlayer) [inline veomni/models/diffusers/ltx2_3/ltx_transformer/modeling_ltx2_3_transformer.py:481]: It seems always from config init model

File: veomni/models/diffusers/ltx2_3/ltx_transformer/modeling_ltx2_3_transformer.py [hunk 10]
```diff
@@ -0,0 +1,510 @@
...
+        return model
+
+
+def compute_ltx2_loss(
+    predictions: list[torch.Tensor],
+    training_targets: list[torch.Tensor],
+    video_loss_masks: list[torch.Tensor] | None,
+    audio_predictions: list[torch.Tensor] | None = None,
+    audio_training_targets: list[torch.Tensor] | None = None,
+    audio_loss_masks: list[torch.Tensor] | None = None,
```
Comment (FoolPlayer) [inline veomni/models/diffusers/ltx2_3/ltx_transformer/modeling_ltx2_3_transformer.py]: audio loss mask is not used

File: veomni/models/diffusers/ltx2_3/ltx_transformer/modeling_ltx2_3_transformer.py [hunk 11]
```diff
@@ -0,0 +1,510 @@
...
+
+        sample_vlm = None
+        if video_loss_masks is not None and i < len(video_loss_masks):
+            sample_vlm = video_loss_masks[i]
+
+        if sample_vlm is not None:
+            sample_vlm = sample_vlm.to(device=prediction.device, dtype=torch.bool)
+            loss_mask = sample_vlm.view(1, 1, F, H, W).float()
+            masked_loss = per_element_loss * loss_mask
+            valid_count = loss_mask.reshape(1, 1, -1).sum(dim=-1).clamp(min=1e-8)
```
Comment (FoolPlayer) [inline veomni/models/diffusers/ltx2_3/ltx_transformer/modeling_ltx2_3_transformer.py]: Maybe add assert batch size dim == real size, or just reshape to batchsize. Now, the size is  always 1, it's OK.

File: veomni/models/diffusers/ltx2_3/ltx_transformer/modeling_ltx2_3_transformer.py (hunk 9 above)
Comment (JimmyMrr) [reply veomni/models/diffusers/ltx2_3/ltx_transformer/modeling_ltx2_3_transformer.py:481]: cls._from_config initializes the model structure randomly using configuration parameters. No weights are loaded here. Weight loading is handled separately by the upper-layer DiTTrainer through FSDP2's load_state_dict.

File: configs/dit/ltx2_av_lora_low_vram.yaml [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+  data_type: diffusion
+  datasets_type: iterable
+  text_keys: text
+  max_seq_len: 8192
+  dataloader:
+    type: native
+    num_workers: 2
+    drop_last: true
+
+  mm_configs:
```
Comment (Coach257) [inline configs/dit/ltx2_av_lora_low_vram.yaml]: These args appear unused. It seems they should be moved to `model.condition_model_cfg`

**Review by Coach257 — APPROVED** [review]
